src/cls/rbd/cls_rbd_types.cc

```diff
@@ -506,22 +510,23 @@ void MirrorImageMap::generate_test_instances(
   bufferlist data;
   data.append(std::string(128, '1'));
 
-  o.push_back(new MirrorImageMap("uuid-123", data));
-  o.push_back(new MirrorImageMap("uuid-abc", data));
+  o.push_back(new MirrorImageMap("uuid-123", ceph_clock_now(), data));
+  o.push_back(new MirrorImageMap("uuid-abc", ceph_clock_now(), data));
 }
 
 bool MirrorImageMap::operator==(const MirrorImageMap &rhs) const {
-  return instance_id == rhs.instance_id && data.contents_equal(data);
+  return instance_id == rhs.instance_id && mapped_time == rhs.mapped_time &&
+    data.contents_equal(data);
 }
 
 bool MirrorImageMap::operator<(const MirrorImageMap &rhs) const {
-  return  instance_id < rhs.instance_id;
+  return instance_id < rhs.instance_id || mapped_time < rhs.mapped_time;
```

Comment:
Needs to be consistently ordered, so it should compare ```instance_id``` first, then if equal compare ```mapped_timed```